features/all-positions/PositionActionsDialog.tsx

```diff
@@ -0,0 +1,583 @@
+        : "None";
+
+    const underCollateralizedPrice =
+      Number(sponsorPosition.collateral) /
+      (Number(sponsorPosition.tokensOutstanding) *
+        parseFloat(fromWei(collReq)));
+
+    const underCollateralizedPercent =
+      ((Number(latestPrice) - Number(underCollateralizedPrice)) /
+        Number(latestPrice)) *
+      100;
+
+    const collBalanceTooLow = () => {
+      if (tabIndex == "deposit") {
+        return collBalanceNum < collateralToDepositNum;
+      }
+      if (tabIndex == "liquidate") {
+        return collBalanceNum < finalFeeNum;
+      }
+    };
+
+    const tokenBalanceTooLow = tokenBalanceNum < maxTokensToLiquidateNum;
+
+    const needCollateralAllowance = () => {
+      if (tabIndex == "deposit") {
+        if (collAllowance === null || collateralToDeposit === null) return true;
+        if (collAllowance === "Infinity") return false;
+        return collAllowance < parseFloat(collateralToDeposit);
+      }
+      if (tabIndex == "liquidate") {
+        if (collAllowance === null || finalFee == null) return true;
+        if (collAllowance === "Infinity") return false;
+        return collAllowance < finalFee;
+      }
+      return true;
+    };
+    const needTokenAllowance = () => {
+      if (tokenAllowance === null || maxTokensToLiquidate === null) return true;
+      if (tokenAllowance === "Infinity") return false;
+      return tokenAllowance < parseFloat(maxTokensToLiquidate);
+    };
+
+    // Liquidation requires both collateral and synthetic to be approved. This
+    // results in a number of different configurations the user could be in depending
+    // on their interaction with this Dapp. Process the text here to keep the
+    // component simple.
+    const liquidationNeedAllowanceText = () => {
+      if (!needCollateralAllowance() && !needTokenAllowance()) {
+        return null;
+      }
+      if (needCollateralAllowance() && !needTokenAllowance()) {
+        return `You will need to sign two transactions one to approve collateral ${collSymbol} and a second to preform the liquidation.`;
+      }
+      if (!needCollateralAllowance() && needTokenAllowance()) {
+        return `You will need to sign two transactions one to approve synthetic ${tokenSymbol} and a second to preform the liquidation.`;
+      }
+      if (needCollateralAllowance() && needTokenAllowance()) {
+        return `You will need to sign three transactions one to approve synthetic ${tokenSymbol}, one to approve collateral ${collSymbol} and a third to preform the liquidation.`;
+      }
+    };
+
+    const executeDeposit = async () => {
+      if (collateralToDeposit && !collBalanceTooLow() && emp) {
+        setHash(null);
+        setSuccess(null);
+        setError(null);
+        const collateralToDepositWei = toWei(collateralToDeposit);
+
+        try {
+          if (needCollateralAllowance()) await setMaxCollateralAllowance();
+          const tx = await emp.depositTo(props.selectedSponsor, [
```

Comment:
What if `props.selectedSponsor` is null?

Reply:
we would not have gotten to this logic as there is the large if check at the top of the rendered component that checks this.